week1/homework/Program.cs

```diff
@@ -0,0 +1,107 @@
+var builder = WebApplication.CreateBuilder(args);
+var app = builder.Build();
+
+app.MapGet("/", () => "Hello World!");
+
+app.MapGet("/info", () => $"hello " + Environment.Version);
+
+
+// Question 1: string manipuation
+app.MapGet("/1", () =>
+{
+  string input = "suman ghimire";
+
+  string reversed = ReverseString(input); //TODO: Implement ReverseString
+
+  string ReverseString(string word)
+  {
+    char[] array = word.ToCharArray();
+    Array.Reverse(array);
+    return new String(array);
+  }
+
+  Console.WriteLine($"Reversed input value: {reversed}");
+  return ($"Reversed input value: {reversed}");
+});
+
+//2. String/Math
+// Return the number (count) of vowels in the given string. Consider a, e, i, o, u as vowels.
+app.MapGet("/2", () =>
+{
+  string input = "IntellectualizationOOYa";
+  int vowelCount = GetVowelCount(input.ToLower()); //TODO: Implement GetVowelCount
+
+  int GetVowelCount(string word)
+  {
+    int count = 0;
+    foreach (char c in word)
+    {
+      if (c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u')
+      {
+        count += 1;
+      }
+    }
+    return count;
+  }
+  Console.WriteLine($"Number of vowels: {vowelCount}");
+  return ($"Number of vowels: {vowelCount}");
+});
+
+//3. Math/Array
+//Given an array of the numbers return an array with two elements where first element represents sum of all negative numbers and second element represents multiplication of all positive numbers;
+
+app.MapGet("/3", () =>
+{
+  int[] arr = new[] { 271, -3, 1, 14, -100, 13, 2, 1, -8, -59, -1852, 41, 5 };
+  int[] result = GetResult(arr); //TODO: Implement GetResult
+  int[] GetResult(int[] array)
+  {
+    int countP = 0;
+    int countN = 1;
```

Comment:
it is generally better to gibe full names for the variables, it makes the code easier to understand.

Here it is a bit confusing:
1. I see the `countP` variable and I assume the `P` is for `positive` based on the task
2. In the implementation below you are doing `if (item < 0) countP += item;` which is the use case if the numbers are `negative`

Reply:
yes i corrected it . thank you